src/hotspot/share/services/memReporter.cpp

```diff
@@ -170,6 +171,12 @@ void MemSummaryReporter::report_summary_of_type(MEMFLAGS flag,
 
   size_t reserved_amount  = reserved_total (malloc_memory, virtual_memory);
   size_t committed_amount = committed_total(malloc_memory, virtual_memory);
+  // There can be up to two CDS archives which can contain readonly data. On Windows, pages are not
+  // shareable so the RO region may not actually be read only
+  size_t read_only_bytes;
+  FileMapInfo::current_info() != nullptr ? read_only_bytes = FileMapInfo::current_info()->readonly_total() : read_only_bytes = 0; // static archive
+  if (FileMapInfo::dynamic_info() != nullptr)
+    read_only_bytes += FileMapInfo::dynamic_info()->readonly_total(); // dynamic archive
```

Comment:
Still does not belong here in this detail... NMT should not have to know that there are multiple FileMapInfo; it should not even have to know what a FileMapInfo is.

Proposal: Merge this into `FileMapInfo::readonly_total()`, make `FileMapInfo::readonly_total()` static, adapt comment (comment above about Windows details belongs atop of the prototype). And here, we just want to have a "const size_t readonly = FileMapInfo::readonly_total".

Ideally, it would be cool had CDS an outward facing minimal header (e.g. cdsStatistics.hpp) with APIs like these (e.g. `static size_t CDS::readonly_total()`. Because we still need to include the rather fat header fileMapInfo.hpp.

Reply:
Thank you for the feedback! I agree that NMT should be blind to these features but I'm not entirely sure the best way to pass on the readonly amount without at least having FileMapInfo. Maybe there should be an RFE for the minimal header and then some cleanup?